src/kangaroo.rs

```diff
@@ -0,0 +1,192 @@
+use crunchy::unroll;
+use super::{PLEN, KeccakFamily, RHO, PI, Permutation};
+
+const RC: [u64; 12] = [
+	0x000000008000808b,
+	0x800000000000008b,
+	0x8000000000008089,
+	0x8000000000008003,
+	0x8000000000008002,
+	0x8000000000000080,
+	0x000000000000800a,
+	0x800000008000000a,
+	0x8000000080008081,
+	0x8000000000008080,
+	0x0000000080000001,
+	0x8000000080008008,
+];
+
+#[allow(unused_assignments)]
+pub fn keccakf(a: &mut [u64; PLEN]) {
+    for i in 0..12 {
+        let mut array: [u64; 5] = [0; 5];
```

Comment:
Can you hoist `array` outside of the loop? Seems that it we initialize it in `theta` section anyway (we just need to add `array[x] = 0` right on the beginning of the loop.
But if the benchmarks don't improve then let's not bother, cause it's more readable this way.